src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -115,16 +121,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request, op
 // pipeData pipes data between the client and proxy connections.  It's also
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, op *ops.Op, closeFunc func()) {
-	writeErrCh := make(chan error, 1)
-	// Start piping from client to proxy
-	op.Go(func() {
-		_, writeErr := io.Copy(connOut, clientConn)
-		writeErrCh <- writeErr
-	})
-
-	// Then start copying from proxy to client.
-	_, readErr := io.Copy(clientConn, connOut)
-	writeErr := <-writeErrCh
```

Comment:
Here's the scenario:

1. Client connects
2. Proxy connects to server
3. Proxy starts piping both ways
4. Client disconnects
5. Reading from the client will stop at this point due to an EOF, but writing to the client doesn't stop, so we keep both connections open

The simplistic way to do this is to just close both connections as soon as either io.Copy is finished, which is what we used to do, but that sometimes results in us closing a connection while we're still doing an io op on it.  The new code in netx sets timeouts on the io ops and interrupts in between ops.